src/components/ChatWindow.tsx

```diff
@@ -76,163 +76,192 @@ export default function ChatWindow() {
   };
 
   const sendMessage = async (msg: string) => {
-
-    const content: Content = {
-      username: username,
-      message: msg
+    if (props.livekitHelper.enterText(msg)) {
+      const content: Content = {
+        username: username,
+        message: msg,
+      };
+      addMessage({
+        content: {
+          username: content.username,
+          content: content.message,
+        },
+        type: MessageType.SEND,
+      });
     }
 
-    const message: Message = {
-      command: Command.SEND_MESSAGE,
-      channel: "general",
-      content: JSON.stringify(content)
-    }
+    // const message: Message = {
+    //   command: Command.SEND_MESSAGE,
+    //   channel: "general",
+    //   content: JSON.stringify(content)
+    // }
 
-    const jsonData = JSON.stringify(message);
+    // const jsonData = JSON.stringify(message);
 
-    socket.send(jsonData);
+    // socket.send(jsonData);
 
-    const messageData: ListElem = {
+    // const messageData: ListElem = {
+    //   content: {
+    //     username: content.username,
+    //     content: content.message
+    //   },
+    //   type: MessageType.SEND
+    // }
+
+    // console.log("Sent Message", jsonData, messageData);
+    // addMessage(messageData);
+  };
+
+  const handleData = (msg: string, participant: string) => {
+    addMessage({
       content: {
-        username: content.username,
-        content: content.message
+        username: participant,
+        content: new TextDecoder().decode(msg),
       },
-      type: MessageType.SEND
-    }
-
-    console.log("Sent Message", jsonData, messageData);
-    addMessage(messageData);
+      type: MessageType.RECEIVE,
+    });
   };
 
   onMount(async () => {
-    let WEBSOCKET_URL : string = process.env.WEBSOCKET_URL!;
-    let fullURL : string = `ws://` + WEBSOCKET_URL +`/chat?username=${username}`;
-    console.debug('connecting to ' + fullURL);
-    // Create WebSocket connection.
-    socket = new WebSocket(fullURL);
-
-
-    // Connection opened
-    socket.addEventListener("open", (event) => {
-      const subscribeMessage = {
-        command: Command.SUBSCRIBE,
-        channel: "general"
-      }
-
-      const jsonData = JSON.stringify(subscribeMessage);
-
-      socket.send(jsonData);
-
-      const messageData: ListElem = {
-        content: {
-          username: "client"!,
-          content: "...Welcome!..."
-        },
-        type: MessageType.SPECIAL_REQUEST
-      }
-
-      addMessage(messageData);
+    window.addEventListener("DATA_RECEIVED", (params: any) => {
+      const { msg, participant } = params.detail;
+      handleData(msg, participant);
     });
-
+    // Instantiate the LiveKitHelper with the server URL, API key, and API secret
```

Comment:
I think we should make a function handle for the livekit / websocket logic for now